### ISSUE TITLE
fatalError when decoding a struct with 16-element array property

### ISSUE THREAD

**Issue by kosyak** [Flight-School/MessagePack#10]
Greetings! I've imported MessagePack via SwiftPM (version 1.2.2) and now playing with this code:

```swift
struct X: Codable {
    let y: [Int]
}

let object = X(y: [Int](repeating: 0, count: 16))
let encoded = try! MessagePackEncoder().encode(object)
let decoded = try! decoder.decode(X.self, from: encoded)
```

The last command gives me 
```
Fatal error: dataCorrupted(Swift.DecodingError.Context(codingPath: [_GenericIndexKey(stringValue: "Index 1", intValue: 1)], debugDescription: "Invalid format: 220", underlyingError: nil)): file /mydir/SourcePackages/checkouts/MessagePack/Sources/MessagePack/Decoder/UnkeyedDecodingContainer.swift, line 49
```

220 is `array 16` in msgpack spec. Are such structs (with array fields that could contain 16+ elements) unsupported by the library?

**Comment by mattt** [comment]
Hey @kosyak, thanks so much for reporting this.

This was indeed a bug. I was able to create a failing test with ea754a0 and resolve it with 0b83a62.

Can you please confirm that your issue is resolved with the latest in `master`?

**Comment by kosyak** [comment]
`master` works without issues for me. Awesome, thank you!

**Comment by mattt** [comment]
So glad to hear it! This change available in version 1.2.3. Thanks again for pointing this out.
